ivy/functional/frontends/numpy/ndarray/ndarray.py

```diff
@@ -249,10 +249,10 @@ def transpose(self, axes, /):
     def swapaxes(self, axis1, axis2, /):
         return np_frontend.swapaxes(self, axis1, axis2)
 
-    def all(self, axis=None, out=None, keepdims=False, *, where=True):
+    def all(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
```

Comment:
Hi @DragosStoican yes that's correct, numpy.any doesn't have any dtype argument and according to [documentation](https://numpy.org/doc/stable/reference/generated/numpy.ndarray.any.html) it seems that ndarray.any also doesn't. But in reality ndarray calls [these](https://github.com/numpy/numpy/blob/main/numpy/core/_methods.py) functions. You can verify this using the following example 
```
axis=None
out=None
keepdims=False
np.array([True,False]).any(axis,out,keepdims)
```
Since we are using positional arguments, its parsed as `axis=axis, dtype=out, out=keepdims`, which cause the error `TypeError: output must be an array `

Reply:
Interesting. I still wouldn't agree with adding the dtype argument to our func signature. The signatures have to be identical with the frontend framework. And it's even more confusing because the dtype argument is not used.

Is this use of positional arguments causing the test to fail?

Reply:
Yes, if keyword arguments were being used there would be no error. But I think adding this is indeed necessary because if we are trying to mirror numpy.ndarray behaviour the function signature has to be exact same. I think the dtype argument at least for `any` and `all` can only accept bool argument, otherwise it should raise an error as casting from bool to int or float will be considered unsafe. So we can possibly include the dtype argument and raise an error based on the function to mirror the exact same behaviour of numpy. Similarly all other functions under ndarray needs to be updated too. Happy to know your opinion @DragosStoican

Reply:
If the test would be calling the original numpy function with keyword arguments instead of positional arguments, then we wouldn't be getting this error correct? I think we should be able to enforce the execution with keyword arguments only

Reply:
Yes that's true that using keyword arguments while testing will avoid error. But in my opinion that would be just a hacky way to avoid the error and not mirroring the framework exactly. So something like this would work in numpy but fail in ivy.
```
import numpy as np
np.array([True,False]).any(0,bool,None, False) #will work

import ivy.functional.frontends.numpy as np
np.array([True,False]).any(0,bool,None, False) #will throw error
```

Also other than this, there was a bug in the treatment of `where` parameter which has been corrected in this PR.
@DragosStoican